apidoc/Titanium/Android/QuickSettingsService.yml

```diff
@@ -0,0 +1,217 @@
+
+  - name: getIcon
+    summary: Returns the Tile's current icon.
+    returns:
+        type: [String, Titanium.Blob, Titanium.Filesystem.File]
+
+  - name: getState
+    summary: Returns the Tile's current state.
+    returns:
+        type: Number
+        constants: Titanium.Android.TILE_STATE_*
+
+  - name: getLabel
+    summary: Returns the Tile's current label
+    returns:
+        type: String
+
+    name: isLocked
+    summary: Returns 'true' if the device is currently locked, 'false' otherwise.
+    returns:
+        type: Boolean
+
+    name: isSecure
+    summary: Returns 'true' if the device is in secure state, 'false' otherwise.
+    returns:
+        type: Boolean
+
+  - name: showDialog
+    summary: Opens an Alert dialog.
+    description: |
+      Creates and shows a default Alert dialog with the provided options dictionary.
+
+      Note: Alert dialog only supports one of the following: message/options. If you pass both - only `message` will
+      be shown.
+    parameters:
+      - name: options
+        type: showParams
+        summary: Dictionary containing the options for the dialog.
+
+  - name: startActivityAndCollapse
+    summary: Colapses the quick settings menu and starts an activity for the passed Intent
+    parameters:
+      - name: intent
+        summary: Intent to be fired
+        type: Titanium.Android.Intent
+
+  - name: unlockAndRun
+    summary: Prompts the user to unlock the device and runs the JS code
+    parameters:
+      - name: jsCode
+        summary: JavaScript code to be evaluated
+        type: String
+
+events:
+  - name: startlistening
+    summary: Tile is listening for events.
+    description: |
+      Dispatched whenever the user has opened the quick setting menu and the Tile is added in there.
+
+  - name: stoplistening
+    summary: Tile has stopped listening for events.
+    description: |
+      Dispatched whenever the user has collapsed the quick setting menu and the Tile is not visible.
+
+  - name: tileadded
+    summary: The Tile has been added in the quick menu.
+    description: |
+      Dispatched wheneved the user has moved the Tile for this service in the quick settings menu.
+
+  - name: tileremoved
+    summary: The Tile has been removed from the quick menu.
+    description: |
+      Dispatched wheneved the user has removed the Tile for this service from the quick settings menu.
+
+  - name: tiledialogoptionselected
+    summary: An item from the signle choice menu has been selected.
+    properties:
+      - name: itemIndex
+        summary: Index of the selected item from the single choice menu in the dialog.
+
+  - name: tiledialogcancelled
+    summary: Dispatched when the alert dialog has been cancelled.
+
+  - name: tiledialogpositive
+    summary: Dispatched when the positive (index 0) button has been clicked.
+
+  - name: tiledialogneutral
+    summary: Dispatched when the neutral (index 1) button has been clicked.
+
+  - name: tiledialognegative
```

Comment:
While I do understand that we also used lowercase-only event naming-conventions, I'd like to discuss to move away from that. We should have camel-cased event-driven event-names, like `tileRemoved` and `tileDialogOptionSelected`. It would also look better on Alloy, e.g. `on="tileRemoved"`. Any downsides?

Reply:
If you are talking about using camel-cased event-names everywhere (not only for this service) we may have some trouble with deprecated event-names which are camel-cased. I had a case where the runtime does not behave consistently when two events with the same name (one lowercased, one camel-cased) are dispatched at the same time.